test/sdp_interop.js

```diff
@@ -549,203 +479,6 @@ a=rtcp-mux\r\n"
     "Not expected Unified Plan output")
 });
 
-QUnit.test('audioInactiveUnifiedPlan2PlanB', function (assert) {
```

Comment:
I'm trying to figure out what's broken with these test cases, but I can't see the problem, could you please help me understand? Is it not possible to modify them and keep them?

Reply:
George those specific testcases didn't make sense to me. The plan-b sdp provided didn't have audio mids in the bundle. This is not possible with the transceiver model since when the initial offer is generated, a receiver will always be present even if there is no source added. So the direction will be "recvonly" and a ssrc will be assigned for the receiver. So we will start off with having both the mids (0, 1) in the bundle line and later we won't be able to take them out.
Also, I tried to modify the 2-way-jtisi and 3-way-jitsi json files but there are far too many changes that I have to do make them pass. I was only able to edit half of them. The main differences between the previous translation logic and this one are
1. m-lines are not generated for 'recvonly' ssrcs in the remote description. It doesn't make sense in the transceiver model since there is always a receiver and an ssrc associated for every sender.
2. a=ssrc with msid attribute is always generated in plan-b to unified plan conversion which wasn't happening before.
3. The order of the a=ssrc lines is maintained which wasn't happening before.

Reply:
> George those specific testcases didn't make sense to me. The plan-b sdp provided didn't have audio mids in the bundle. This is not possible with the transceiver model since when the initial offer is generated, a receiver will always be present even if there is no source added. So the direction will be "recvonly" and a ssrc will be assigned for the receiver. So we will start off with having both the mids (0, 1) in the bundle line and later we won't be able to take them out.

Jitsi Meet doesn't have such a use case, but is it generally impossible to only have inactive audio tracks in the unified plan? How should the plan-b sdp look like in that case?

> Also, I tried to modify the 2-way-jtisi and 3-way-jitsi json files but there are far too many changes that I have to do make them pass. I was only able to edit half of them.

The playbooks capture two main flows: a complete 2-way call and a complete 3-way call (from call join to media exchange). It's okay if there are a lot of changes or if we need to completely rewrite them, but I think it's important to keep them (and even add more, if possible).

Feel free to change the existing playbook format (that is used to represent a call flow) if you find it cumbersome to work with it.

Reply:
> > George those specific testcases didn't make sense to me. The plan-b sdp provided didn't have audio mids in the bundle. This is not possible with the transceiver model since when the initial offer is generated, a receiver will always be present even if there is no source added. So the direction will be "recvonly" and a ssrc will be assigned for the receiver. So we will start off with having both the mids (0, 1) in the bundle line and later we won't be able to take them out.
> 
> Jitsi Meet doesn't have such a use case, but is it generally impossible to only have inactive audio tracks in the unified plan? How should the plan-b sdp look like in that case?
> 
It is possible to have inactive audio tracks but the bundle line will have the mids of the inactive audio tracks as they would start as "recvonly" first and then be marked as "inactive" later. I will modify the bundle line and bring it back in.
> > Also, I tried to modify the 2-way-jtisi and 3-way-jitsi json files but there are far too many changes that I have to do make them pass. I was only able to edit half of them.
> 
> The playbooks capture two main flows: a complete 2-way call and a complete 3-way call (from call join to media exchange). It's okay if there are a lot of changes or if we need to completely rewrite them, but I think it's important to keep them (and even add more, if possible).
> 
> Feel free to change the existing playbook format (that is used to represent a call flow) if you find it cumbersome to work with it.

Ok, I will rewrite them if I find it cumbersome to edit each one of those descriptions.

Reply:
Thanks Jaya.